src/store/messagesStore.js

```diff
@@ -111,6 +111,24 @@ const state = {
 }
 
 const getters = {
+	/**
+	 * Returns whether more messages can be loaded, which means that the current
+	 * message list doesn't yet contain all future messages.
+	 * If false, the next call to "lookForNewMessages" will be blocking/long-polling.
+
+	 * @param {object} state the state object.
+	 * @param {object} getters the getters object.
+	 * @returns {bool} true if more messages exist that needs loading, false otherwise
+	 */
+	hasMoreMessagesToLoad: (state, getters) => (token) => {
+		const conversation = getters.conversation(token)
+		if (!conversation) {
+			return false
+		}
+
+		return getters.getLastKnownMessageId(token) < conversation.lastMessage.id
```

Comment:
So this can be inaccurate, as conversation.lastMessage is not always updated.
But I guess you only notice this when there are only such system messages on the "new page" of the chat. So let's ignore that for now.